### PR TITLE
assignment 3 jan (no dependencies)

### PR DESCRIPTION
**Pull request by jvdassen**
This is my testing where no framework/library was used, please tell me if you find anything :)

### CODE REVIEW

File: src/edu/stanford/nlp/net/Ports.java [hunk 1]
```diff
@@ -29,6 +29,19 @@ public static boolean available(int port) {
       ss.setReuseAddress(true);
       ds = new DatagramSocket(port);
       ds.setReuseAddress(true);
+
+      int ssPort = ss.getLocalPort();
+      int dsPort = ds.getLocalPort();
+      boolean osPickedSpecifiedPort = port == ssPort && port == dsPort;
+      if(!osPickedSpecifiedPort) {
```
Comment (RinorSefa) [inline src/edu/stanford/nlp/net/Ports.java]: Good insight

File: src/edu/stanford/nlp/util/StringUtils.java [hunk 2]
```diff
@@ -1738,13 +1739,16 @@ public static String toTitleCase(String s) {
    *         false otherwise
    */
   public static boolean isTitleCase(String s) {
-    String[] words = s.split("\\s+");
-    for (String w : words) {
-      if (!(Character.isUpperCase(w.charAt(0)) && (w.substring(1).toLowerCase().equals(w.substring(1))))) {
-        return false;
-      }
+    Predicate<String> capitalized = word -> Character.isUpperCase(word.charAt(0));
```
Comment (RinorSefa) [inline src/edu/stanford/nlp/util/StringUtils.java]: I had to google this, nice usage of Predicate

File: src/edu/stanford/nlp/util/StringUtils.java [hunk 3]
```diff
@@ -1738,13 +1739,16 @@ public static String toTitleCase(String s) {
...
+    Predicate<String> remainderIsLowerCase = w -> w.substring(1).toLowerCase().equals(w.substring(1));
+
+    Predicate<String> titleCased = capitalized.and(remainderIsLowerCase);
```
Comment (RinorSefa) [inline src/edu/stanford/nlp/util/StringUtils.java]: Let me know if I got this right, so the capitalized  and remainderIsLowerCase are like methods which are merged now in titleCased, and then the assesment is happenign in the .allMatch(titlecase) method

File: test/src/edu/stanford/nlp/util/StringUtilsTest.java [hunk 4]
```diff
@@ -3,10 +3,36 @@
...
+import java.util.function.ToLongFunction;
 import java.util.regex.Pattern;
+import java.util.stream.BaseStream;
+import java.util.stream.Collector;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
 
 public class StringUtilsTest {
```
Comment (RinorSefa) [inline test/src/edu/stanford/nlp/util/StringUtilsTest.java]: WOW, great job

**Review by RinorSefa — APPROVED** [review]
Great, code quality. I learned something from doing this code review.

File: src/edu/stanford/nlp/util/StringUtils.java (hunk 3 above)
Comment (jvdassen) [reply src/edu/stanford/nlp/util/StringUtils.java]: Yes, you understood it absolutely right! It basically does the same thing as the previous code did - but in my opinion more readable and for very large strings it has better performance

File: src/edu/stanford/nlp/net/Ports.java (hunk 1 above)
Comment (jvdassen) [reply src/edu/stanford/nlp/net/Ports.java]: Thanks, Rinor

File: test/src/edu/stanford/nlp/math/SloppyMathTest.java [hunk 5]
```diff
@@ -32,28 +32,28 @@ public void testRound2() {
...
   public void testMin() {
-    assertEquals(1,SloppyMath.min(1,2,3));
+    assertEquals(1, SloppyMath.min(1, 2, 3));
   }
 
   public void testIsDangerous() {
     assertTrue(SloppyMath.isDangerous(Double.POSITIVE_INFINITY) &&
-                 SloppyMath.isDangerous(Double.NaN) &&
-                 SloppyMath.isDangerous(0));
+            SloppyMath.isDangerous(Double.NaN) &&
```
Comment (julwil) [inline test/src/edu/stanford/nlp/math/SloppyMathTest.java]: @jvdassen  indentation. Maybe already done :)

**Review by julwil — APPROVED** [review]
Looks good to me, good job

**Pull request closed without merging**